runbot/migrations/17.0.5.8/post-migration.py

```diff
@@ -0,0 +1,123 @@
+
+    # first_seen_build_id was not stored, lets fill it and update all values for good mesure
+    for error, vals in vals_by_error.items():
+        cr.execute('UPDATE runbot_build_error_content SET first_seen_build_id = %s, first_seen_date = %s, last_seen_build_id = %s, last_seen_date = %s WHERE id=%s', (vals[0], vals[1], vals[2], vals[3], error))
+
+    # generate flattened error hierarchy
+    cr.execute('''SELECT
+                    id,
+                    parent_id
+                FROM runbot_build_error_content
+                ORDER BY id
+                ''')
+
+    error_by_parent = {}
+    for error_id, parent_id in cr.fetchall():
+        if parent_id:
+            error_by_parent.setdefault(parent_id, []).append(error_id)
+    stable = False
+    while not stable:
+        stable = True
+        for parent, child_ids in error_by_parent.items():
+            for child_id in child_ids:
+                if parent == child_id:
+                    continue
+                sub_childrens = error_by_parent.get(child_id)
+                if sub_childrens:
+                    error_by_parent[parent] = error_by_parent[parent] + sub_childrens
+                    error_by_parent[child_id] = []
+                    stable = False
+    for parent, child_ids in error_by_parent.items():
+        if parent in child_ids:
+            _logger.info('Breaking cycle parent on %s', parent)
+            error_by_parent[parent] = [c for c in child_ids if c != parent]
+            cr.execute('UPDATE runbot_build_error_content SET parent_id = null WHERE id=%s', (parent,))
+    error_by_parent = {parent: chilren for parent, chilren in error_by_parent.items() if chilren}
+
+    cr.execute('''SELECT
+                    id,
+                    active,
+                    parent_id
+                    random,
+                    content,
+                    test_tags,
+                    tags_min_version_id,
+                    tags_max_version_id,
+                    team_id,
+                    responsible,
+                    customer,
+                    fixing_commit,
+                    fixing_pr_id
+                FROM runbot_build_error_content
+                WHERE parent_id IS null
+                ORDER BY id
+                ''')
+    errors = cr.fetchall()
+    nb_groups = len(error_by_parent)
+    _logger.info('Creating %s errors', nb_groups)
+    for error in errors:
+        error_id, *values = error
+        children = error_by_parent.get(error_id, [])
+        assert not error_id in children
+        all_errors = [error_id, *children]
+        error_count = len(all_errors)
+
+        # vals_by_error order: min(build_id), min(log_date), max(build_id), max(log_date)
+        build_count = 0
+        first_seen_build_id = first_seen_date = last_seen_build_id = last_seen_date = None
+        if error_id in vals_by_error:
+            error_vals = [vals_by_error[error_id] for error_id in all_errors]
```

Comment:
```suggestion
            error_vals = [vals_by_error[error_id] for error_id in all_errors if error_id in vals_by_error]
```
To avoid script failure when to the error has no build linked.